### ISSUE TITLE
Removing selected annotation needs undocummented call of cancelSelected before deletion

### ISSUE THREAD

**Issue by AlexMili** [annotorious/annotorious#221]
Hello,

Thank you for the great work you did on this amazing lib !

I needed to delete annotation when pressing backspace or delete key, so I implemented what you proposed in #193:
```javascript
let anno = new Annotorious({
    image: document.getElementById("image"),
    allowEmpty: true,
    disableEditor: true,
    drawOnSingleClick: true,
});

anno.setDrawingTool("polygon");

document.addEventListener('keyup', (event) => {
    if (event.key == "Backspace" || event.key == "Delete") {
        var selected = anno.getSelected();
        if (selected != null) {
            anno.removeAnnotation(selected);
        }
    }
});
```

When I use this code, I get an error saying that `this.selectedShape is null` when I try to create a new annotation after the deletion of the previous one. I fixed it by adding `anno.cancelSelected()` before `anno.removeAnnotation(selected)`, resulting in:
```javascript
document.addEventListener('keyup', (event) => {
    if (event.key == "Backspace" || event.key == "Delete") {
        var selected = anno.getSelected();
        if (selected != null) {
            anno.cancelSelected();
            anno.removeAnnotation(selected);
        }
    }
});
```

The error is also present in the [Headless mode guide](https://recogito.github.io/guides/headless-mode/).

**Comment by rsimon** [comment]
Thanks for this - I tested in the headless mode demo and think I can see what's going on. (A small bug, with a missing null-check.) Just to confirm: the second solution works for you as a workaround now, right?

**Comment by AlexMili** [comment]
Yes the second solution works for me :)

**Comment by AlexMili** [comment]
Thought it was good but I just found a case in which it doesn't work.
Right after creating my annotation, the annotation is automatically selected. When I press the delete key at this moment (without clicking outside annotation after its creation), the annotation disappears but as soon as I click again on the image, the annotation appears again and the `createSelection` is fired again.

**Comment by rsimon** [comment]
Ok, thanks for testing this out. I'll need to reproduce myself to be sure. But my hunch is that it has to do with the fact that the annotation isn't a "proper annotation" at that moment yet. It's a "selection" object, which doesn't yet have an ID assigned - which then probably messes things up. The fix is probably easy, but definitely needs tweaks inside Annotorious somewhere. I'll investigate!

**Comment by rsimon** [comment]
Hi again,

I added a fix that definitely resolves the first issue (needed call of `cancelSelected`, as well as the error shown when creating a new annotation in headless mode).

I _think_ the fix should also solve your second issue. But I'm not 100% sure I reproduced the right setting. In my test I used headless mode, and then left the annotation "open". (I did not call `saveSelected` inside of the `createSelection` handler.) I could then delete the annotation normally.

The fix will go live with the next release, 2.7.4. Let me know if that fixes your issue.

Cheers,
R

**Comment by AlexMili** [comment]
Hi,
It works fine if I add `saveSelected` inside `createSelection` handler but does not work otherwise. It looks like it is working (no error) but when you try to create a new annotation, the newly created anno disappears and the previously deleted one (if it exists) appears again.

Using `saveSelected` is totally fine so It's Ok for me :)
